frontend/www/js/omegaup/components/arena/NavbarProblems.vue

```diff
@@ -35,12 +35,20 @@
         >
           <span class="mr-1">{{ getMaxScoreForProblem(problem) }}</span>
           <font-awesome-icon
-            v-if="problem.bestScore == problem.maxScore"
+            v-if="
+              (typeof problem.myBestScore !== 'undefined' &&
+                problem.myBestScore == problem.maxScore) ||
+              (typeof problem.myBestScore == 'undefined' &&
+                problem.bestScore == problem.maxScore)
+            "
             icon="check"
             :style="{ color: 'green' }"
           />
           <font-awesome-icon
-            v-else-if="problem.hasRuns"
+            v-else-if="
+              (typeof problem.hasMyRuns !== 'undefined' && problem.hasMyRuns) ||
+              (typeof problem.hasMyRuns == 'undefined' && problem.hasRuns)
+            "
```

Comment:
Mismo caso, esto se puede simplificar de la siguiente forma:
```suggestion
            v-else-if="problem.hasMyRuns || problem.hasRuns"
```

Reply:
Solo que ahí mas bien tendria que ser algo como v-else-if="problem.hasMyRuns || problem.hasRuns && problem.hasMyRuns === null"  Porque si no va a tomar el hasruns de otros concursantes siendo administrador o incluso al concursante en el modo práctica.

Reply:
Solo que ahí mas bien tendria que ser algo como v-else-if="problem.hasMyRuns || problem.hasRuns && problem.hasMyRuns === null"  Porque si no va a tomar el hasruns de otros concursantes siendo administrador o incluso al concursante en el modo práctica.

Reply:
Tienes razón. Me confundí un poco con las condiciones que tenías, pero parece mejor como sugieres